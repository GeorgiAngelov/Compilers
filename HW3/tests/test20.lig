type a : [a];

var x : a;

var y : a = [x];

function main(argv : [[int]]) : int { }
