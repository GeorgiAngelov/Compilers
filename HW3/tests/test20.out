(decls
   (decl-type:ok (id a) (type-array (type-id a)))
   (decl-var:ok (id x) (type-id a))
   (decl-var:ok (id y) (type-id a) (array-lit:[a] (exps (id:a x))))
   (decl-fun:ok (id main) (type-fun (params (typed-id (id argv) (type-array (type-array (type-int))))) (type-int))))
